src/NuGetForUnity/Editor/PackageSource/NugetPackageSourceV3.cs

```diff
@@ -200,35 +198,51 @@ public INugetPackage GetSpecificPackage(INugetPackageIdentifier package)
             string versionConstraints = "")
         {
             var packagesToFetch = packages as IList<INugetPackage> ?? packages.ToList();
-
             var packagesFromServer = Task.Run(
                     async () =>
                     {
-                        var updates = new List<INugetPackage>();
-                        for (var i = 0; i < packagesToFetch.Count;)
+                        if (SupportsPackageIdSearchFilter)
                         {
-                            var searchQueryBuilder = new StringBuilder();
-                            for (var inner = 0; inner < UpdateSearchBatchSize && i < packagesToFetch.Count; inner++, i++)
+                            var updates = new List<INugetPackage>();
+                            for (var i = 0; i < packagesToFetch.Count;)
                             {
-                                if (i > 0)
+                                var searchQueryBuilder = new StringBuilder();
+                                for (var inner = 0; inner < UpdateSearchBatchSize && i < packagesToFetch.Count; inner++, i++)
                                 {
-                                    searchQueryBuilder.Append(' ');
+                                    if (i > 0)
+                                    {
+                                        searchQueryBuilder.Append(' ');
+                                    }
+
+                                    searchQueryBuilder.Append($"packageid:{packagesToFetch[i].Id}");
```

Comment:
If you are using string interpolation here then perhaps a cleaner solution instead of StringBuilder is to just build a List<string> and then use `string.Join(" ", list)`.

Reply:
We would need:
```
var query = string.Join(' ', packagesToFetch.Skip(i).Take(UpdateSearchBatchSize).Select(package => $"{package.Id}_"));
i += UpdateSearchBatchSize;
```
As there is no Chunk method in older C# we need the index. Not sure if the Unity implementation of the `Skip` already has optimization for `List`. Do you think it would make the code more readable, with the Skip / Take?

Another think is we could use `AppendFormat` as none .Net Core doesn't has the `Append` overload that takes a `AppendInterpolatedStringHandler`.

Reply:
I just thought to replace StringBuilder with List<string> and ToString with string.join leaving everything else the same.